File: C4_InnerTube/InnerTubeCS/InnerTube/ConverterImage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows.Data;
using System.IO;
using SharedUtilities;
using System.Windows.Media;

namespace InnerTube
{
    [ValueConversion(typeof(object), typeof(string))]
    public class ConverterImage : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            string path = (string)value;
            return SetImage(path);           
        }

        public static object SetImage(string imagePath)
        {
            //if the path doesn't exist
            if (String.IsNullOrEmpty(imagePath) || (!File.Exists(imagePath)))
            {
                //use default image
                return FileHelper.DefaultImage;
            }
            else
            {
                return imagePath;
            }
        }


        public object ConvertBack(object value, Type targetType,
            object parameter, System.Globalization.CultureInfo culture)
        {
            // we don't intend this to ever be called
            return null;
        }
    }
}

File: C4_InnerTube/InnerTubeVB/InnerTube/ConverterImage.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Windows.Data
Imports System.IO
Imports SharedUtilities


Namespace InnerTube
    <ValueConversion(GetType(Object), GetType(String))> _
    Public Class ConverterImage
        Implements IValueConverter
        Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.Convert
            Dim path As String = CStr(value)
            Return SetImage(path)
        End Function

        Public Shared Function SetImage(ByVal imagePath As String) As Object
            'if the path doesn't exist
            If String.IsNullOrEmpty(imagePath) OrElse ((Not File.Exists(imagePath))) Then
                'use default image
                Return Path.Combine(App.Settings.SubPath, FileHelper.DefaultImage)
            Else
                Return imagePath
            End If
        End Function

        Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.ConvertBack
            ' we don't intend this to ever be called
            Return Nothing
        End Function
    End Class




End Namespace
